src/test/shared/schema/schemaAssertions.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { JSONObject, unmarshal } from './schemaTypes'
+
+/**
+ * Assert whether or not name exists under definitionName in the JSON schema
+ * @schema the JSON schema
+ * @param definitionName The name of the definition to use
+ * @param name The name of the property to look for
+ * @returns void if the test passes otherwise fails the assertion
+ */
+export function assertDefinitionProperty(schema: JSONObject, definitionName: string, name: string) {
+    const definitionProperties = unmarshal(schema, 'definitions', definitionName, 'properties')
+    assertProperty(definitionProperties, name)
+}
+
+/**
+ * Assert whether name exists at an arbitary location in the JSON schema
+ * @param arbitrarySchemaLocation An arbitary location in the JSON schema
+ * @param name The name of the property to look for
+ * @returns void if the test passes otherwise fails the assertion
```

Comment:
This is personal preference but I'd just mark the return type as `void | never` instead of the docstring.